src/shaders/SkColorShader.cpp

```diff
@@ -128,6 +171,29 @@ SkShader::GradientType SkColorShader::asAGradient(GradientInfo* info) const {
     return kColor_GradientType;
 }
 
+SkSpotColorShader::SkSpotColorShader(const SkString& spotColorName) : fSpotColorName{spotColorName} {}
+
+sk_sp<SkFlattenable> SkSpotColorShader::CreateProc(SkReadBuffer& buffer) {
+    SkString spotColorName;
+    buffer.readString(&spotColorName);
+    return sk_make_sp<SkSpotColorShader>(spotColorName);
+}
+
+void SkSpotColorShader::flatten(SkWriteBuffer& buffer) const {
+    buffer.writeString({fSpotColorName.c_str(), fSpotColorName.size()});
+}
+
+SkShader::GradientType SkSpotColorShader::asAGradient(GradientInfo* info) const {
```

Comment:
I don't quite get for what `asAGradient` is used but seems to be needed and looks reasonable.

Reply:
Yep, I just followed the pattern established by `SkColorShader` which is just a flat color. Just like `SkColorShader`, `SkSpotColorShader` returns `kColor_GradientType` which represents a flat color "gradient".